sympy/core/mul.py

```diff
@@ -1545,37 +1564,48 @@ def _eval_is_extended_negative(self):
         return self._eval_pos_neg(-1)
 
     def _eval_is_odd(self):
-        is_integer = self.is_integer
-
-        if is_integer:
+        from sympy import trailing, fraction
+        if self.is_integer:
+            n, d = fraction(self)
+            if d.is_Integer and d.is_even:
+                # if minimal power of 2 in num vs den is
+                # positive then we have an even number
+                if (Add(*[i.as_base_exp()[1] for i in
+                        Mul.make_args(n) if i.is_even]) - trailing(d.p)
+                        ).is_positive:
+                    return False
             r, acc = True, 1
             for t in self.args:
-                if not t.is_integer:
-                    return None
-                elif t.is_even:
+                assert t.is_integer
+                if t.is_even:
+                    return False
+                if r is False:
+                    pass
+                elif acc != 1 and (acc + t).is_odd:
                     r = False
-                elif t.is_integer:
-                    if r is False:
-                        pass
-                    elif acc != 1 and (acc + t).is_odd:
-                        r = False
-                    elif t.is_odd is None:
-                        r = None
+                elif t.is_odd is None:
```

Comment:
Is there a way that `t.is_odd` is different from `fuzzy_not(t.is_even)`?

Reply:
If `e` is even then `(2/e).is_even` is False but `(2/e).is_odd` is None.

Reply:
But in the context of this code, `t.is_integer` is True and `denom(t) == 1` so I don't think there would be a difference. Why?

Reply:
Since you tested for `t.is_even`, I was just thinking about if one could draw some conclusions based on that result and not have to evaluate `t.is_odd`, which may or may not be time consuming.